bitnami/cassandra/templates/_helpers.tpl

```diff
@@ -180,3 +180,75 @@ Return true if a TLS credentials secret object should be created
 {{- end -}}
 {{- end -}}
 
+{{/*
+Returns the available value for certain key in an existing secret (if it exists),
+otherwise it generates a random value.
+*/}}
+{{- define "getValueFromSecret" }}
+    {{- $len := (default 16 .Length) | int -}}
+    {{- $obj := (lookup "v1" "Secret" .Namespace .Name).data -}}
```

Comment:
I have my concerns with this approach, as it forces a cluster to be available. In the past using helm template caused errors, could you confirm that this approach is also compatible with other deployment approaches like helm template + kubectl?

Reply:
Sure, I can use this on an offline laptop to successfully execute the `helm template`. It will generate a random secret, as if there is no existing one. 

According to the official helm document：

> Helm is not supposed to contact the Kubernetes API Server during a helm template or a helm install|update|delete|rollback --dry-run, so the lookup function will return an empty list (i.e. dict) in such a case.

So, Yes, It will compatible with `helm template` + `kubectl` and any other deployment approaches.